pkg/agent/flexvolume/types.go

```diff
@@ -34,9 +34,11 @@ type VolumeManager interface {
 type AttachOptions struct {
```

Comment:
Help me better understand why we are using a single struct that has both block and filesystem options stuffed into it.  Is there something that prevents us from using a dedicated type for each?

Reply:
No reason to use a single struct. Since we are using the same pluginfor both block and file and kubernetes is providing us with the same struct, i thought we could reuse the same struct for filesystem. I could separate into a different structs as well.

Reply:
Actually i dont think this can be done. You have to use the same struct since Kubernetes is giving you that as input. The input is being unmarshalled here https://github.com/rook/rook/blob/master/cmd/rookflex/cmd/mount.go#L50. So at this point, we dont know whether this is a block or a cephfs attachment request.